besu/src/main/scripts/unixStartScript.txt

```diff
@@ -182,8 +182,18 @@ APP_ARGS=`save "\$@"`
 # Collect all arguments for the java command, following the shell quoting and substitution rules
 eval set -- \$DEFAULT_JVM_OPTS \$JAVA_OPTS \$${optsEnvironmentVar} <% if ( appNameSystemProperty ) { %>"\"-D${appNameSystemProperty}=\$APP_BASE_NAME\"" <% } %>-classpath "\"\$CLASSPATH\"" <% if ( mainClassName.startsWith('--module ') ) { %>--module-path "\"\$MODULE_PATH\"" <% } %>${mainClassName} "\$APP_ARGS"
 
-# limit malloc to 2 arenas, and use jemalloc if available
-export MALLOC_ARENA_MAX=2
-export LD_PRELOAD=libjemalloc.so
+if [ "\$darwin" = "false" -a "\$msys" = "false" ]; then
+  # check if jemalloc is available
+  TEST_JEMALLOC=\$(LD_PRELOAD=libjemalloc.so sh -c true 2>&1)
+
+  # if jemalloc is available the output is empty, otherwise the output has an error line
+  if [ -z "\$TEST_JEMALLOC" ]; then
+    export LD_PRELOAD=libjemalloc.so
+  else
+    echo "INFO: jemalloc not found, you could improve memory usage installing it."
```

Comment:
I'm not sure that echoing from the shell script is a good option for logging. It won't integrate with any logging configuration options and so won't necessarily be written to the right place. We could check if the `LD_PRELOAD` env var is unset on the java side to log this warning via the logging system.

Reply:
Good point, I was not totally sure of this output as well, and I agree that moving the log to java side is a stronger solution.

Reply:
it is worth noting that LD_PRELOAD is primarily a linux thing.  If we move this check into java, we are going to have to derive what our OS is so we don't erroneously report missing lib.  E.g. mac or windows

Reply:
moved the log from the script to Java, and used a dedicated env var when using jemalloc, since `LD_PRELOAD` could be used for other things